### PR TITLE
[Frontend] Multimodal support in offline chat

### LINKED ISSUE #8117: [Doc]: `LLM.chat()` docstring incorrectly suggests multiple chats can be generated in one call
### 📚 The doc issue

According to the docstring for the LLM class's `chat()` method,

> **messages** – A list of messages to generate responses for. Each message is a list of dictionaries with ‘role’ and ‘content’ keys.

This implies `messages` can be a list containing multiple lists, each of which is one chat conversation with the dictionaries for role and content. However, that's not the case: `messages` must be a list of dictionaries, not a list of lists of dictionaries, because only one conversation is supported. See this example: https://github.com/vllm-project/vllm/issues/6416#issuecomment-2318413123

### Suggest a potential alternative/fix

Two options:

1. The docstring is edited to say "A list of messages to generate responses for. Each message is a dictionary with ‘role’ and ‘content’ keys."  Under Returns, say something like "A list containing one RequestOutput object with the generated response."
2. Make the function actually do what the docstring says. If you have multiple conversations you want responses to, doing them in a batch is much more efficient than doing them one at a time by repeatedly calling `chat()`. In my application, I need 12,000 chat responses, and calling `chat()` 12,000 times will take something like 12 hours, whereas calling `generate()` once with the pre-tokenized prompts takes an hour or two.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR follows up to #5049 by adding support for multi-modal inputs in `LLM.chat`. To do so, I implemented a sync version of the chat parser.

cc @petersalas since I further refactored your parsing logic

Also fixes #8117

### CODE REVIEW

File: vllm/transformers_utils/tokenizers/mistral.py [hunk 1]
```diff
@@ -52,12 +52,13 @@ def __init__(self, tokenizer: PublicMistralTokenizer) -> None:
         assert isinstance(self.tokenizer,
                           (Tekkenizer, SentencePieceTokenizer)), type(
                               self.tokenizer)
-        self._is_tekken = isinstance(self.tokenizer, Tekkenizer)
 
-        if self._is_tekken:
+        if (is_tekken := isinstance(self.tokenizer, Tekkenizer)):
```
Comment (petersalas) [inline vllm/transformers_utils/tokenizers/mistral.py:56]: Did you mean to include this change?

**Review by petersalas — APPROVED** [review]
LGTM

File: vllm/transformers_utils/tokenizers/mistral.py (hunk 1 above)
Comment (DarkLight1337) [reply vllm/transformers_utils/tokenizers/mistral.py:56]: Yes, this fixes a typing issue which failed mypy locally.

**Review by simon-mo — APPROVED** [review]
Stamp on behalf of Roger

**Pull request merged**